### PR TITLE
Use 'dev' version on the master branch

### PR DESCRIPTION
**Pull request by ArthurHoaro**
Allowed check branches are now `latest` and `stable`.
Add an updateMethod to match the current remote branch for updates

Relates to #777 and #778

### CODE REVIEW

**Comment by nodiscc** [comment]
So what is the process to do a release with a fixed version number?

Eg when 0.9.0 is released I imagine that would be:

 * Set the version in `index.php`, `shaarli_version.php` to `0.9.0`
 * Tag this commit `0.9.0` and `latest`, do a Github release
 * Next commit should change back the version to `dev`

Is that right?

File: index.php [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <?php
 /**
- * Shaarli v0.8.3 - Shaare your links...
+ * Shaarli dev - Shaare your links...
```
Comment (virtualtam) [inline index.php]: not needed if `master` is no longer used for version checks

File: index.php [hunk 2]
```diff
@@ -25,7 +25,7 @@
 /*
  * PHP configuration
  */
-define('shaarli_version', '0.8.2');
+define('shaarli_version', 'dev');
```
Comment (virtualtam) [inline index.php]: could be read from `shaarli_version.php`?

**Comment by ArthurHoaro** [comment]
Nope. The wiki needs to be updated to be exhaustive but basically:

  - create `v0.9` branch if it doesn't exist
  - merge `master` into `v0.9`
  - set `0.9.0` version in file and push it to `v0.9`
  - tag and release from `v0.9` branch
  - merge `v0.9` branch into `latest`

The `v0.x` branch is here just in case we need to release a new `v0.8.y` version. `v0.x` branches one digit behind the `stable` branch can be deleted.

File: index.php (hunk 1 above)
Comment (ArthurHoaro) [inline index.php]: Yes, also not needed at all to be honest.

File: index.php (hunk 2 above)
Comment (ArthurHoaro) [inline index.php]: That's do-able. I probably need to import other files first, but that will limit the number of manual operation during the release.

**Comment by ArthurHoaro** [comment]
Updated regarding @virtualtam comments.

**Pull request merged**